### DISCUSSION TITLE
Instructions to build free.rm.skytube.oss_??.apk

### DISCUSSION THREAD

**Discussion by KlaasVogel** [SkyTubeTeam/SkyTube#1284] [Q&A] [answered]
Hi @gzsombor,
nice job updating the app to V2.989.
quick question (I hope it's appropriate to drop it here):

I can only run SkyTube (not extra) on my Android TV. I tried using "Android Studio" to build a new apk, but it only generates an apk for SkyTube-Extra. Can you point me to any instructions to build the apk for free.rm.skytube.oss_??.apk ?
If I can build them myself, I might try to solve issue https://github.com/SkyTubeTeam/SkyTube/issues/1211 (no experience in working with Android Studio, so I'm not sure I'll succeed).

with kind regards,
Klaas

**Comment by mckaygerhard** [comment]
i m just curious.. why you dont use skytube-extra? you said that you can build and generate it.. the extra is also open the difference is that "extra" uses some libraries with non gpl compatible licence as i know

**Reply by mckaygerhard** [reply]
there also https://github.com/SkyTubeTeam/SkyTubeLegacy for older devices.. too

**Reply by KlaasVogel** [reply]
hi @mckaygerhard ,

thanx for your reply. When I try to use Skytube(legacy)-extra on my TV, it shows a white screen for a couple of ms and then exits. I just tried a couple of versions (including legacy) and the bare skytube version seems to be the most stable solution. I might have to accept that Sony won't release any software updates and I am glad that at least the "basic" version works. on my TV.

**Comment by gzsombor** [comment, marked as answer]
Android Studio (and Gradle) can be a bit pain, I totally agree :)
You can list most of the build tasks with ```gradle tasks```.
I regularly call ```gradle assembleOssDebug``` to build the OSS debug version, and run 
```~/apps/android-sdk-linux_x86/platform-tools/adb install -r  app/build/outputs/apk/oss/debug/SkyTube-Oss-2.989.apk``` to install it on a locally running emulator.

**Reply by KlaasVogel** [reply]
hi @gzsombor, thanx for your reply. Gradle is just a mystery/black box for me at the moment, including 'gradle tasks', but I will try out your suggestion immediately. Working with adb is fortunately no problem: I use it to install/test the app directly on my TV
